src/engine/strat_engine/mdv.rs

```diff
@@ -159,26 +186,30 @@ impl MetadataVol {
     pub fn filesystems(&self) -> EngineResult<Vec<FilesystemSave>> {
         let mut filesystems = Vec::new();
 
-        for dir_e in read_dir(self.mount_pt.join(FILESYSTEM_DIR))? {
-            let dir_e = dir_e?;
+        with_mount(&self.dev.devnode(), &self.mount_pt, || {
+            for dir_e in read_dir(self.mount_pt.join(FILESYSTEM_DIR))? {
+                let dir_e = dir_e?;
 
-            if dir_e.path().ends_with(".temp") {
-                continue;
-            }
+                if dir_e.path().ends_with(".temp") {
+                    continue;
+                }
 
-            let mut f = OpenOptions::new().read(true).open(&dir_e.path())?;
-            let mut data = Vec::new();
-            f.read_to_end(&mut data)?;
+                let mut f = OpenOptions::new().read(true).open(&dir_e.path())?;
+                let mut data = Vec::new();
+                f.read_to_end(&mut data)?;
 
-            filesystems.push(serde_json::from_slice(&data)?);
-        }
+                filesystems.push(serde_json::from_slice(&data)?);
+
+
+            }
+            Ok(())
+        })?;
 
         Ok(filesystems)
     }
 
     /// Tear down a Metadata Volume.
     pub fn teardown(self, dm: &DM) -> EngineResult<()> {
```

Comment:
Should the ```umount``` here be kept? Does fs need to be unmounted for teardown to succeed?

Reply:
The assumption is that the MDV is unmounted between calls, so it should already be unmounted when teardown is called.

Reply:
It might not be. The related PR talks about an optimization to sometimes not unmount if busy, and a previous unmount could have failed.